src/main/java/ai/shreds/Application/ApplicationProductServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ApplicationProductServiceImpl implements ApplicationProductServicePort {
+
+    private final DomainProductServicePort domainProductService;
+
+    @Override
+    public SharedProductDTO getProductDetails(String id) {
+        try {
+            log.info("Retrieving product details for ID: {}", id);
+            DomainProductEntity productEntity = domainProductService.getProductDetails(id);
+            return productEntity.toDTO();
+        } catch (NotFoundException e) {
+            log.error("Product not found with ID: {}", id, e);
+            throw new ProductNotFoundException("Product not found with id: " + id, e);
+        }
+    }
+
+    @Override
+    public SharedProductDTO createProductDetails(SharedCreateProductRequest request) {
+        validateProductData(request);
+        log.info("Creating product with name: {}", request.getName());
+        DomainProductEntity productEntity = DomainProductEntity.fromRequest(request);
+        productEntity = domainProductService.createProductDetails(productEntity);
+        return productEntity.toDTO();
+    }
+
+    @Override
+    public SharedProductDTO updateProductDetails(String id, SharedUpdateProductRequest request) {
+        validateProductData(request);
+        try {
+            log.info("Updating product with ID: {}", id);
+            DomainProductEntity productEntity = DomainProductEntity.fromUpdateRequest(request);
+            productEntity = domainProductService.updateProductDetails(id, productEntity);
+            return productEntity.toDTO();
+        } catch (NotFoundException e) {
+            log.error("Product not found with ID: {}", id, e);
+            throw new ProductNotFoundException("Product not found with id: " + id, e);
+        }
+    }
+
+    @Override
+    public void deleteProductDetails(String id) {
+        try {
+            log.info("Deleting product with ID: {}", id);
+            domainProductService.deleteProductDetails(id);
+        } catch (NotFoundException e) {
+            log.error("Product not found with ID: {}", id, e);
+            throw new ProductNotFoundException("Product not found with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<SharedProductDTO> listProducts(SharedListProductsParams params) {
+        log.info("Listing products with filters: {}", params);
+        List<DomainProductEntity> productEntities = domainProductService.listProducts(params);
+        return productEntities.stream().map(DomainProductEntity::toDTO).collect(Collectors.toList());
+    }
+
+    private void validateProductData(Object request) {
+        if (request instanceof SharedCreateProductRequest) {
+            SharedCreateProductRequest createRequest = (SharedCreateProductRequest) request;
+            if (createRequest.getPrice() <= 0) {
+                throw new InvalidProductDataException("Product price must be greater than zero.");
```

Comment:
Create the InvalidProductDataException and add a handler for this in the AdapterExceptionHandler